components/autofill/src/sync/address/incoming.rs

```diff
@@ -3,21 +3,25 @@
 * file, You can obtain one at http://mozilla.org/MPL/2.0/.
 */
 
-use super::{Record, RecordData};
+use super::AddressRecord;
 use crate::error::*;
+use crate::sync::{MergeResult, RecordImpl};
 use interrupt_support::Interruptee;
 use rusqlite::{named_params, types::ToSql, Connection};
 use sql_support::ConnExt;
 use sync15::Payload;
 use sync_guid::Guid as SyncGuid;
 use types::Timestamp;
 
-type IncomingState = crate::sync::IncomingState<RecordData>;
-type IncomingAction = crate::sync::IncomingAction<RecordData>;
+type IncomingState = crate::sync::IncomingState<AddressRecord>;
+type IncomingRecordInfo = crate::sync::IncomingRecordInfo<AddressRecord>;
+type LocalRecordInfo = crate::sync::LocalRecordInfo<AddressRecord>;
+type IncomingAction = crate::sync::IncomingAction<AddressRecord>;
 
 /// The first step in the "apply incoming" process for syncing autofill address records.
 /// Incoming tombstones will saved in the `temp.addresses_tombstone_sync_staging` table
 /// and incoming records will be saved to the `temp.addresses_sync_staging` table.
+// XXX - will end up moving to the Impl trait??
```

Comment:
Is this still a consideration or is the placement of `stage_incoming` finalized?